src/Microsoft.AspNetCore.Http.Features/FeatureReferences.cs

```diff
@@ -26,34 +26,53 @@ public FeatureReferences(IFeatureCollection collection)
         public TFeature Fetch<TFeature, TState>(
             ref TFeature cached,
             TState state,
-            Func<TState, TFeature> factory)
+            Func<TState, TFeature> factory) where TFeature : class
         {
-            var cleared = false;
-            if (Revision != Collection.Revision)
+            if (Revision == 0)
```

Comment:
This makes some assumptions about the implementation of Collection.Revision and the initial value. E.g. WebListener's Revision count starts at 0 so this code will keep executing.